File: src/main/java/com/summer/lijiahao/utils/OpenNCConfigAction.java
package com.summer.lijiahao.utils;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.ui.Messages;
import com.summer.lijiahao.abs.AbstractAnAction;
import com.summer.lijiahao.base.NccEnvSettingService;

import java.io.File;

public class OpenNCConfigAction extends AbstractAnAction {
    @Override
    public void doAction(AnActionEvent event) {
        try {
            String ncHome = NccEnvSettingService.getInstance().getNcHomePath();
            if (ncHome == null || ncHome.equals("")) {
                throw new Exception("未配置NC home");
            }

            String configDirPath = ncHome + File.separatorChar + "bin" + File.separatorChar;
            if (isWindows()) {
                configDirPath = configDirPath + "sysConfig.bat";
            } else {
                configDirPath = configDirPath + "sysConfig.sh";
            }

            File configFile = new File(configDirPath);
            if (!configFile.isFile()) {
                throw new Exception("配置文件不存在");
            }

            Runtime.getRuntime().exec(configDirPath);
        } catch (Exception e) {
            String message = e.getMessage();
            Messages.showInfoMessage(message, "Error");
        }
    }

    public boolean isWindows() {
        return System.getProperty("os.name").toLowerCase().contains("windows");
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/menu/action/OpenNCConfigAction.kt
package com.summer.lijiahao.nccdevtool.menu.action

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ui.Messages
import com.summer.lijiahao.nccdevtool.base.action.AbstractAnAction
import com.summer.lijiahao.nccdevtool.devconfig.service.NCCloudEnvSettingService
import java.io.File
import java.util.*

class OpenNCConfigAction : AbstractAnAction() {
    override fun doAction(event: AnActionEvent) {
        try {
            val ncHome: String = NCCloudEnvSettingService.getInstance(event).ncHomePath
            if (ncHome.isEmpty()) {
                throw Exception("未配置NC home")
            }
            var configDirPath = ncHome + File.separatorChar + "bin" + File.separatorChar
            configDirPath = if (isWindows) {
                configDirPath + "sysConfig.bat"
            } else {
                configDirPath + "sysConfig.sh"
            }
            val configFile = File(configDirPath)
            if (!configFile.isFile) {
                throw Exception("配置文件不存在")
            }
            Runtime.getRuntime().exec(configDirPath)
        } catch (e: Exception) {
            val message = e.message
            Messages.showInfoMessage(message, "Error")
        }
    }

    private val isWindows: Boolean
        get() = System.getProperty("os.name").lowercase(Locale.getDefault()).contains("windows")
}
